### PR TITLE
Added error-reference entry #4012

### PR DESCRIPTION
**Pull request by jasnow**
Added error-reference entry #4012 - It is not there yet.

https://sorbet.org/docs/error-reference#4012

I welcome suggestions and corrections.

### Motivation

It was missing.

### Test plan
 
Documentation only.

### CODE REVIEW

File: website/docs/error-reference.md [hunk 1]
```diff
@@ -120,6 +120,18 @@ class Main
...
+This error occurs when an object is defined as class and a method.
+An example is ActiveStorage::Attachment where it is defined
+as "class ActiveStorage::Attachment < ActiveRecord::Base" in
+
+ * sorbet/rails-rbi/models/active_storage/attachment.rbi
+ 
+and module ("def attachment; end") in
+
+ * sorbet/rbi/sorbet-typed/lib/activestorage/all/activestorage.rbi
+
```
Comment (jez) [inline website/docs/error-reference.md]: Can you replace your example with this:

https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end

and include both a code block and a link to the sorbet.run?

Comment (jez) [reply website/docs/error-reference.md]: Ideally, the error-reference should give some sort of troubleshooting  advice for how to fix the error.

Comment (jez) [reply website/docs/error-reference.md]: Actually, there are more things that could trigger this error:

https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end%0A%0Aclass%20C%3B%20end%0AC%20%3D%201%0A%0AD%20%3D%201%0Aclass%20D%3B%20end

Comment (jasnow) [reply website/docs/error-reference.md]: Just added your suggestions. Thanks

File: website/docs/error-reference.md [hunk 2]
```diff
@@ -120,6 +120,36 @@ class Main
...
+Example with class:
+```ruby
+class A; end
+module A; end
+
+module B; end
+class B; end
+````
+[Sorbet.run link](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end)
```
Comment (jez) [inline website/docs/error-reference.md]:
```suggestion
[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end)
```

File: website/docs/error-reference.md [hunk 3]
```diff
@@ -120,6 +120,36 @@ class Main
...
+module B; end
+class B; end
+
+class C; end
+C = 1
+
+D = 1
+class D; end
+```
+[Sorbet.run link](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end%0A%0Aclass%20C%3B%20end%0AC%20%3D%201%0A%0AD%20%3D%201%0Aclass%20D%3B%20end)
```
Comment (jez) [inline website/docs/error-reference.md]:
```suggestion
[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end%0A%0Aclass%20C%3B%20end%0AC%20%3D%201%0A%0AD%20%3D%201%0Aclass%20D%3B%20end)
```

File: website/docs/error-reference.md [hunk 4]
```diff
@@ -120,6 +120,36 @@ class Main
...
+```ruby
+class A; end
+module A; end
+
+module B; end
+class B; end
+````
+[Sorbet.run link](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%3B%20end%0Amodule%20A%3B%20end%0A%0Amodule%20B%3B%20end%0Aclass%20B%3B%20end)
+  
+Example with constant:
```
Comment (jez) [inline website/docs/error-reference.md]: Sorry it wasn't clear: the second example (this one) includes everything from the previous example. We can keep just this one example and delete the other one.

**Comment by jasnow** [comment]
@jez - Did my 3rd change

**Comment by jez** [comment]
> @jez - Did my 3rd change

Thanks! Looks like the URL changed but the code block didn't.

**Comment by jasnow** [comment]
Looks good to me.

**Comment by jasnow** [comment]
Unclear how to make the build system happy, so closing this PR.

**Pull request closed without merging**